src/config.rs

```diff
@@ -128,20 +130,90 @@ pub struct StarshipConfig {
 impl StarshipConfig {
     /// Initialize the Config struct
     pub fn initialize() -> Self {
-        if let Some(file_data) = Self::config_from_file() {
-            StarshipConfig {
-                config: Some(file_data),
+        let mut config = toml::value::Table::new();
+
+        if let Some(global_config_path) = Self::global_config_path() {
```

Comment:
Most of this logic appears to be repeated.
Could we perhaps create an iterator of directories to `.fold()` through with a merging function?